src/cpp/lib/QtWidgets/QTableWidgetItem/qtablewidgetitem_wrap.cpp

```diff
@@ -56,22 +56,35 @@ Napi::Object QTableWidgetItemWrap::init(Napi::Env env, Napi::Object exports) {
 QTableWidgetItem* QTableWidgetItemWrap::getInternalInstance() {
   return this->instance;
 }
-QTableWidgetItemWrap::~QTableWidgetItemWrap() { delete this->instance; }
+QTableWidgetItemWrap::~QTableWidgetItemWrap() {
+    if (!this->disableDeletion) {
+	delete this->instance;
+    }
+}
 
 QTableWidgetItemWrap::QTableWidgetItemWrap(const Napi::CallbackInfo& info)
     : Napi::ObjectWrap<QTableWidgetItemWrap>(info) {
   Napi::Env env = info.Env();
   Napi::HandleScope scope(env);
 
-  if (info.Length() == 1) {
-    QString text =
-        QString::fromUtf8(info[0].As<Napi::String>().Utf8Value().c_str());
-    this->instance = new QTableWidgetItem(text);
-  } else if (info.Length() == 0) {
-    this->instance = new QTableWidgetItem();
+    
+  if (info.Length() > 0 && info[0].IsExternal()) {
+    // --- if external ---
+    this->instance = info[0].As<Napi::External<QTableWidgetItem>>().Data();
+    if (info.Length() == 2) {
+      this->disableDeletion = info[1].As<Napi::Boolean>().Value();
+    }
   } else {
-    Napi::TypeError::New(env, "Wrong number of arguments")
-        .ThrowAsJavaScriptException();
+    if (info.Length() == 1) {
+      QString text =
+          QString::fromUtf8(info[0].As<Napi::String>().Utf8Value().c_str());
+      this->instance = new QTableWidgetItem(text);
+    } else if (info.Length() == 0) {
+      this->instance = new QTableWidgetItem();
+    } else {
+      Napi::TypeError::New(env, "QTableWidgetItemWrap: Wrong number of arguments")
```

Comment:
Just a FYI: Not sure if you want this (I prefer it, as it points to this specific) ---  I just noticed that I left in my extra info to trace down which object was throwing the "Wrong number of arguments" as every single wrapper appears to have the same error message.

Reply:
:+1: 

For new code I've been leaving that message out. If it does trigger then its always an internal bug in NodeGui. Applications don't call this part of the system directly.

Also, at some stage this constructor will need to be rewritten to support the other combinations of arguments which the C++ counterpart has.